src/Mono.Android/Android.Runtime/ResourceIdManager.cs

```diff
@@ -31,12 +31,19 @@ public static void UpdateIdValues ()
 			}
 		}
 
-		[UnconditionalSuppressMessage ("Trimming", "IL2026", Justification = "Types in Resource.designer.cs are preserved, because it is the root assembly passed to the linker.")]
+		[return: DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
 		static Type? GetResourceTypeFromAssembly (Assembly assembly)
 		{
+			const string rootAssembly = "Types in Resource.designer.cs are preserved, because it is the @(TrimmerRootAssembly).";
+
+			[UnconditionalSuppressMessage ("Trimming", "IL2026", Justification = rootAssembly)]
+			[UnconditionalSuppressMessage ("Trimming", "IL2073", Justification = rootAssembly)]
+			[return: DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
+			static Type AssemblyGetType (Assembly a, string name) => a.GetType (name);
```

Comment:
Does this really work? I can't see how the trimmer would be able to determine what the returned type is and so it won't be able to preserve the public methods on it. I think at the very least the `name` would need to be attributed as well, but I'm not sure if that's enough.

Reply:
No, I don't think the trimmer preserves anything here -- this is just ignoring the warning.

This code is looking for a `Resource.designer.cs` file that is generated in every app, which is the root assembly. There will likely be some problem here for NativeAOT one day.

Reply:
I did preserve the attribute that is kind of the "entry point":
```csharp
[AttributeUsage (AttributeTargets.Assembly)]
public class ResourceDesignerAttribute : Attribute
{
	public ResourceDesignerAttribute (string fullName)
	public ResourceDesignerAttribute (
			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
			string fullName)
	{
		FullName = fullName;
	}

	[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
	public string FullName { get; set; }
```

Reply:
Could we add a `Type` property to the attribute?
```c#
[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)] public Type Type { get; set; }
```

That would be perfectly trimmable and AOT friendly.

Reply:
We can add a new API in .NET 9, but currently Xamarin.Android class libraries built against `Mono.Android.dll` from ~2012 *still work*. This is different from iOS, as they have breaking changes.

So there would be some number of libraries out there that won't use the new API.

Reply:
This PR is *for* .NET 9,so we can certainly add the suggested `Type` property, *and also* have appropriate fallback code for when `Type` is `null`.  (This in turn means that the new property would need to allow `null`, e.g. `Type? Type {get;set;}`.) We'd also want a "semantically meaningful" name; what is `ResourceDesignerAttribute.Type` supposed to mean?  A better name is warranted.  `ResourceDesignerType`, perhaps?

```csharp
[assembly: Android.Runtime.ResourceDesignerAttribute(
    "Example.Android.Resource",
    ResourceDesignerType=typeof(Example.Android.Resource),
    IsApplication=true
)]
```

That said… the "new" Resource design of dc3ccf28cdbe9f8c0a705400b83c11a85c81a980 was thought/hoped to (somewhat?) address trimming support; by using properties/methods for everything the trimmer should be able to "see" what methods are used and preserve them, while allowing *everything else* to be trimmed away.

Setting `[DynamicallyAccessedMembers-(DynamicallyAccessedMemberTypes.PublicMethods)]` feels contrary to that intention?  Why would *all* methods on the `Resource` type need to be preserved?

Rando aside: why'd we use a *string* to name the `Resource` type instead of `typeof()` in the first place?  A constructor overload feels reasonable here!

```csharp
partial class ResourceDesignerAttribute {
    public ResourceDesignerAttribute (Type resourceDesignerType);
}
```

Reply:
> Setting [DynamicallyAccessedMembers-(DynamicallyAccessedMemberTypes.PublicMethods)] feels contrary to that intention? Why would all methods on the Resource type need to be preserved?

We used System.Reflection to find an `UpdateIdValues()` method and call it, so I think the current trimmer design is to add `DynamicallyAccessedMemberTypes.PublicMethods`?

> partial class ResourceDesignerAttribute {
    public ResourceDesignerAttribute (Type resourceDesignerType);
}

For this part, let me see what it would take to:
1. Generate usage of this overload, as we create `Resource.designer.cs` at build time
2. Maybe the old overload could emit a trimmer warning.

Reply:
@jonathanpeppers [wrote](https://github.com/xamarin/xamarin-android/pull/8758/files#r1508008689):

> We used System.Reflection to find an `UpdateIdValues()` method and call it,

Yes, we *used* to.  Commit https://github.com/xamarin/xamarin-android/commit/dc3ccf28cdbe9f8c0a705400b83c11a85c81a980 should have removed that call for most (all?) cases.  Yes, `UpdateIdValues()` still exists, but "nothing" should call it (unless some customer decided to manually call it For Reasons™?).  *We* certainly shouldn't call it, and we should be *actively **removing** it*, e.g. in `ClearDesignerClass()`:

https://github.com/xamarin/xamarin-android/blob/main/src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/LinkDesignerBase.cs#L137

@dellis1972: do we even *need* `[assembly: ResourceDesigner(…)]` anymore?

Reply:
@dellis1972 [replies on Discord](https://discord.com/channels/732297728826277939/732297837953679412/1212862407421198438):

> Its not called if they are using the new Designer Assembly. If they turn it off then it will be used.

I overlooked/forgot that the new stuff can be disabled.  It's controlled via `$(AndroidUseDesignerAssembly)`.

When can we drop support for this?  It was added in .NET 8; removing support for `$(AndroidUseDesignerAssembly)`=false from .NET 9 feels too early, maybe?  Maybe .NET 10?

Reply:
I'm on with .NET 10 , people should have found all the bugs in the new stuff by then :P